src/main/scala/viper/gobra/frontend/Desugar.scala

```diff
@@ -1952,13 +1968,66 @@ object Desugar {
           }
 
           definedMethods += (proxy -> mem)
-          additionalMembers ::= mem
+          AdditionalMembers.addMember(mem)
         }
       }
     }
     var registeredInterfaces: Set[String] = Set.empty
 
-    var additionalMembers: List[in.Member] = List.empty
+
+
+    object AdditionalMembers {
+
+      private var additionalMembers: List[in.Member] = List.empty
+      private var finalized: Boolean = false
+      private var computationsBeforeFinalize: List[() => Unit] = List.empty
+
+      def addMember(m: in.Member): Unit = additionalMembers ::= m
+      def addFinalizingComputation(f: () => Unit): Unit = computationsBeforeFinalize ::= f
+      def finalizedMembers: List[in.Member] = {
+        require(!finalized)
+
+        if (!finalized) {
+          computationsBeforeFinalize.foreach(_())
+          finalized = true
+        }
+
+        additionalMembers
+      }
+    }
+
+
+    def registerDomain(t: Type.DomainT, dT: in.DomainT): Unit = {
+      if (!registeredDomains.contains(dT.name) && info == t.context.getTypeInfo) {
```

Comment:
Is `info == t.context.getTypeInfo` a sanity check? Shouldn't an exception be thrown if `info` does not match?

Reply:
No, that is not a sanity check. This function can be called from the context of another file.